app/models/course/assessment.rb

```diff
@@ -31,6 +31,8 @@ class Course::Assessment < ApplicationRecord
 
   belongs_to :tab, inverse_of: :assessments
 
+  belongs_to :monitor, class_name: Course::Monitoring::Monitor.name, dependent: :destroy, optional: true
```

Comment:
Having dependent destroy here may not work when there is no dependent destroy heartbeat in session model. When an assessment is destroyed, it will trigger monitor deletion. I understand that there is dependent destroy for session as well.

When rails tries to remove session, it wont be successful since there is no dependent destroy on the heartbeat. Heartbeat has `session_id` as FK to course_monitoring_sessions and when a session is deleted, the session_id FK points to nothing and postgres will prevent it.

I think it's better to remove dependent destroy here. Next time we want to clean the db, we work our way from course_monitoring_monitors table.